Score filter

jakubzaba opened the pull request:
Closes #200 

It goes with options _[any, 6+, 7+, 8+, 9+]_ same as Booking.com.

~**Done, but based on #205, you can review last commit.**~
- [x] ~**Merge after [Gitlab MR #585]**~
- [x] ~**Merge after [Gitlab MR #589]**~

---
Please check this before merging (do not delete this checklist):

- [x] it works in landscape and portrait mode
- [x] it uses `cacheConfig.force=true` if offline access doesn't make sense

mrtnzlml commented:
It doesn't work very well yet:

![jan-23-2018 11-56-00](https://user-images.githubusercontent.com/978368/35258812-817c79ba-0034-11e8-9038-2de9daecf1f0.gif)

jakubzaba commented:
It's the bug I fixed in slider lib, but I always get old code even after I deleted it from `node_nodules`. Do you how to upgrade to current master? I thought I just do sth wrong, but obviously you have the same problem.

mrtnzlml commented:
What version of the library is correct?

jakubzaba commented:
It's defined as `"@ptomasroos/react-native-multi-slider": "ptomasroos/react-native-multi-slider"` and they didn't updated the version since Aug.

mrtnzlml commented:
Ok, we should rebase this after #207. They did not release the new version yet...

VaclavSir commented on app/hotels/src/filter/score/ScoreFilter.js:
```diff
@@ -0,0 +1,57 @@
+
+  getTitle = (minScore: number | null) =>
+    minScore
+      ? rating[minScore - 6] // Minimal score is 6
```
This is a bit cryptic.

I would rename `rating` to `ratingTitles` and changed it to an object (`const ratingTitles = { 6: 'pleasant 6+', 7: 'good 7+', ... }`). Then this method could look like `return ratingTitles[minScore] || 'rating'`.

VaclavSir commented on app/hotels/src/filter/score/ScorePopup.js:
```diff
@@ -0,0 +1,65 @@
+  };
+
+  componentWillReceiveProps = ({ minScore }: Props) =>
+    this.setState({ minScore: minScore ? minScore - SLIDER_SHIFT : 0 }); // Normalize null/6-9 options to slider's range 0-4
```
Instead of the comment, you can introduce a well named function. Something like convertScoreToSliderRange.

VaclavSir commented on app/hotels/src/filter/score/ScorePopup.js:
```diff
@@ -0,0 +1,65 @@
+  onSave = () =>
+    this.props.onSave(
+      this.state.minScore
+        ? this.state.minScore + SLIDER_SHIFT // Normalize back to null/6-9 minScore options from slider's range 0-4
```
Similarly here.

VaclavSir commented on app/hotels/src/filter/score/ScorePopup.js:
```diff
@@ -0,0 +1,65 @@
+
+export default class ScorePopup extends React.Component<Props, State> {
+  state = {
+    minScore: 0,
```
This name is misleading, because this variable contains the slider value, ie. not the score, but the shifted score. Not sure how to rename it, maybe shiftedScore or sliderPosition.

VaclavSir commented on app/hotels/src/filter/score/ScorePopup.js:
```diff
@@ -0,0 +1,65 @@
+        : null,
+    );
+
+  renderLabel = (minScore: number) => {
```
Again, this `minScore` is not the score, it is the slider position.
